modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -182,6 +238,175 @@ MechanicalContactConstraint::jacobianSetup()
+    if (!pinfo || pinfo->_node->n_comp(_sys.number(), _vars[_component]) < 1)
+      continue;
+
+    const Real distance = pinfo->_normal * (pinfo->_closest_point - _mesh.nodeRef(slave_node_num));
+
+    if (pinfo->isCaptured())
+    {
+
+      if (contactResidual < std::abs(distance))
+        contactResidual = std::abs(distance);
+
+      // penetration < tol
+      if (contactResidual > _al_penetration_tolerance)
+      {
+        _console << "Augmented Lagrangian contact penetration enforcement is NOT satisfied \n";
+        converged = 1;
+        break;
+      }
+
+      if (_model == CM_COULOMB)
+      {
+        RealVectorValue contact_force_normal((pinfo->_contact_force * pinfo->_normal) *
+                                             pinfo->_normal);
+        RealVectorValue contact_force_tangential(pinfo->_contact_force - contact_force_normal);
+
+        RealVectorValue tangential_inc_slip =
+            pinfo->_incremental_slip - (pinfo->_incremental_slip * pinfo->_normal) * pinfo->_normal;
+
+        const Real tan_mag(contact_force_tangential.norm());
+        const Real tangential_inc_slip_mag = tangential_inc_slip.norm();
+
+        RealVectorValue distance_vec =
+            (pinfo->_normal * (_mesh.nodeRef(slave_node_num) - pinfo->_closest_point)) *
+            pinfo->_normal;
+
+        Real penalty = getPenalty(*pinfo);
+        RealVectorValue pen_force_normal =
+            penalty * distance_vec + pinfo->_lagrange_multiplier * pinfo->_normal;
+
+        // Frictional capacity
+        Real capacity(_friction_coefficient * (pen_force_normal * pinfo->_normal < 0
+                                                   ? -pen_force_normal * pinfo->_normal
+                                                   : 0.0));
+
+        // incremental slip <= tol for all pinfo_pair such that tan_mag < capacity
+        if (MooseUtils::absoluteFuzzyLessThan(tan_mag, capacity) &&
+            pinfo->_mech_status == PenetrationInfo::MS_STICKING)
+        {
+          if (MooseUtils::absoluteFuzzyGreaterThan(tangential_inc_slip_mag,
+                                                   _al_incremental_slip_tolerance))
+          {
+            _console << "Augmented Lagrangian contact tangential sliding enforcement is NOT "
+                        "satisfied \n";
+            converged = 2;
+            break;
+          }
+        }
+
+        // for all pinfo_pair, tag_mag < (1 + tol) * (capacity + tol)
+        if (MooseUtils::absoluteFuzzyGreaterThan(tan_mag,
+                                                 (1 + _al_frictional_force_tolerance) *
+                                                     (capacity + _al_frictional_force_tolerance)))
+        {
+          _console
+              << "Augmented Lagrangian contact frictional force enforcement is NOT satisfied \n";
+          converged = 3;
+          break;
+        }
+      }
+    }
+  }
+
+  _communicator.max(converged);
+
+  if (converged > 0.0)
+    return false;
```

Comment:
You could print the appropriate message based on the value of `converged` here, and then it would go to processor 0's output.